src/main/java/com/gachigang/ontherun/config/handler/ApplicationExceptionHandler.java

```diff
@@ -1,88 +1,72 @@
 package com.gachigang.ontherun.config.handler;
 
-import com.gachigang.ontherun.common.ExceptionMessage;
-import com.gachigang.ontherun.model.dto.ErrorDetails;
-import jakarta.servlet.http.HttpServletRequest;
-import jakarta.validation.ConstraintViolationException;
+import com.gachigang.ontherun.common.exception.*;
+import com.gachigang.ontherun.model.dto.ErrorMessage;
+import io.jsonwebtoken.JwtException;
+import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ExceptionHandler;
-import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
-import java.time.LocalDateTime;
-
 /**
  * Exception handler for application.
  */
+@Slf4j
 @RestControllerAdvice
 @RequiredArgsConstructor
-@Slf4j
+@PropertySource("classpath:message.properties")
 public class ApplicationExceptionHandler extends ResponseEntityExceptionHandler {
 
-    private final ExceptionMessage exceptionMessage;
+    private final ExceptionWebHandler exceptionWebHandler;
+    private final Environment environment;
 
     /**
-     * Handle all other exceptions.
+     * Handle AuthenticationExceptions.
      */
-    @ExceptionHandler(Exception.class)
-    public ResponseEntity<?> handleAllExceptions(Exception exception, HttpServletRequest request) {
-        log.error("Exception was thrown due unknown exception:", exception);
-
-        ResponseStatus responseStatus =
-                exception.getClass().getAnnotation(ResponseStatus.class);
-        HttpStatus status =
-                responseStatus != null ? responseStatus.value() : HttpStatus.INTERNAL_SERVER_ERROR;
-
-        ErrorDetails message = ErrorDetails.builder()
-                .status(status.value())
-                .timestamp(LocalDateTime.now())
-                .description(exceptionMessage.getUnauthorizedErrorMessage())
-                .url(request.getRequestURL().toString())
-                .build();
-        return new ResponseEntity<>(message, status);
+    @ExceptionHandler(JwtException.class)
+    public ResponseEntity<ErrorMessage> jwtException(@NonNull final AuthenticationException exception) {
```

Comment:
Where this AuthenticationException came from? If it is your custom exception it is not right. You should recieve here JwtException as a root Jwt exception

Reply:
Fixed